src/app/main/main.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-main',
+  templateUrl: './main.component.html',
+  styles: [` `]
+})
+export class MainComponent implements OnInit {
+
+  public randomNumber:any="Button";
+  uservalue = 10 ;
+  uservalues: any ;
+  public child:number
+
+  public onNumberGenerated(randomNumber) {
+      console.log(randomNumber.name)
+      console.log(randomNumber.value)
+      this.randomNumber = randomNumber.name;
+      this.child = randomNumber.value;
+    }
+
+    submit(){
+    const  numberone =  parseInt('uservalue')  + parseInt('child') 
+     if (this.randomNumber == 'inc') {
```

Comment:
Here default there won't be any operation, on inc we need to increment and only if dec then only we need to decrement.